RevenueCatUI/Templates/Components/TiersComponentView.swift

```diff
@@ -0,0 +1,185 @@
+                    selectorFound = true
+                }
+            }
+
+            return !selectorFound
+        }
+
+        return selectorFound ? before : []
+    }
+
+    private var componentAfterSelector: [PaywallComponent] {
+        var selectorFound = false
+        let after = tiers[selectedTierIndex].components.filter { component in
+            if selectorFound == false {
+                if case .tierSelector(_) = component {
+                    selectorFound = true
+                } else if case .tierToggle(_) = component {
+                    selectorFound = true
+                }
+            }
+
+            return selectorFound
+        }
+
+        return selectorFound ? after : tiers[selectedTierIndex].components
+    }
+
+    var tierSelector: PaywallComponent.TierSelectorComponent? {
+        return tiers[selectedTierIndex].components.compactMap { component in
+            if case .tierSelector(let tierSelectorView) = component {
+                return tierSelectorView
+            }
+
+            return nil
+        }.first
+    }
+
+    var tierToggle: PaywallComponent.TierToggleComponent? {
+        return tiers[selectedTierIndex].components.compactMap { component in
+            if case .tierToggle(let tierToggle) = component {
+                return tierToggle
+            }
+
+            return nil
+        }.first
+    }
+
+    var body: some View {
+        VStack(spacing: 0) {
+            ComponentsView(
+                locale: locale,
+                components: self.componentBeforeSelector,
+                configuration: self.configuration
+            )
+
+            if let tierSelector {
+                TierSelectorComponentView(
+                    locale: locale,
+                    component: tierSelector,
+                    tiers: tiers,
+                    selectedTierIndex: $selectedTierIndex
+                )
+            } else if let tierToggle {
+                TierToggleComponentView(
+                    locale: locale,
+                    component: tierToggle,
+                    tiers: tiers,
+                    selectedTierIndex: $selectedTierIndex
+                )
+            } else {
+                TierSelectorComponentView(
+                    locale: locale,
+                    component: .init(displayPreferences: self.component.displayPreferences),
+                    tiers: tiers,
+                    selectedTierIndex: $selectedTierIndex
+                )
+            }
+
+            ComponentsView(
+                locale: locale,
+                components: self.componentAfterSelector,
+                configuration: self.configuration
+            )
+        }
+    }
+
+    struct TierSelectorComponentView: View {
+
+        let locale: Locale
+        let component: PaywallComponent.TierSelectorComponent
+        let tiers: [PaywallComponent.TiersComponent.TierInfo]
+        @Binding var selectedTierIndex: Int
```

Comment:
I see `TierToggleComponentView` has all these private and an internal constructor, not sure if we want to do that here as well